### PR TITLE
updated ballot_name -> name, per Google specs (#219)

### LINKED ISSUE #219: Change "ballot_name" in Party class to "name"
Currently the sample manifests contains the property `name` for a `party`, but the `Party` class in Python contains the property called `ballot_name` and is hashing against this property. If you click on the [link to the Google specs for "Party"](https://developers.google.com/elections-data/reference/party),  it also uses `Name` as the reference... so the Python class here is incorrect and needs to change to match the specs and call it the property `name` (like in the example manifests). 

image

Note: this also implies changes that will ripple to the sample generator as well as to the electionguard-web-api's sample data.

### PR DESCRIPTION
**Pull request by markman123**
### Issue
*Link your PR to an issue*

Fixes #219 

### Description
Updates references of `ballot_name` to the updated and correct `name`

### Testing
Determine that all references to `ballot_name` are now `name`

### Checklist
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] 🤔 **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] ✅ **DO** check open PR's to avoid duplicates.
- [x] ✅ **DO** keep pull requests small so they can be easily reviewed.
- [ ] ✅ **DO** build locally before pushing.
- [ ] ✅ **DO** make sure tests pass.
- [x] ✅ **DO** make sure any new changes are documented.
- [x] ✅ **DO** make sure not to introduce any compiler warnings.
- [x] ❌**AVOID** breaking the continuous integration build.
- [x] ❌**AVOID** making significant changes to the overall architecture.

💚Thank you!

### CODE REVIEW

**Comment by keithrfung** [comment]
@markman123 Looking good. This just needs `pipenv run black .` to pass the linter.

**Comment by markman123** [comment]
> 
> 
> @markman123 Looking good. This just needs `pipenv run black .` to pass the linter.

I did do that again: 
image

is there a different config?

**Comment by keithrfung** [comment]
@markman123 Sorry on the late response. I'm fairly certain this is an issue with the version of `black` installed. I've had this happen before. There are newer versions where it will autocorrect to a different format. We should lock the version to prevent this issue. I'm running version `19.10b0`. I'm guessing the GitHub Action has grabbed latest. Which version are you running?

**Comment by keithrfung** [comment]
https://github.com/microsoft/electionguard-python/pull/235 Should resolve the build issues on this PR.

File: src/electionguard/publish.py [hunk 1]
```diff
@@ -64,13 +64,13 @@ def publish(
 
     make_directory(BALLOTS_DIR)
     for ballot in ciphertext_ballots:
-        ballot_name = BALLOT_PREFIX + ballot.object_id
```
Comment (keithrfung) [inline src/electionguard/publish.py:67]: These instances do not need to be changed. These are unrelated to `Party`. It just happens to be named `ballot_name` but it is the name of the file.

File: tests/integration/test_end_to_end_election.py [hunk 2]
```diff
@@ -459,17 +459,17 @@ def verify_results(self) -> None:
 
         ciphertext_ballots: List[CiphertextAcceptedBallot] = []
         for ballot in self.ballot_store.all():
-            ballot_name = BALLOT_PREFIX + ballot.object_id
```
Comment (keithrfung) [inline tests/integration/test_end_to_end_election.py:462]: These instances do not need to be changed. These are unrelated to Party. It just happens to be named ballot_name but it is the name of the file.

**Review by keithrfung — CHANGES_REQUESTED** [review]

**Comment by keithrfung** [comment]
@markman123 Can you take a look at requested changes?

**Review by keithrfung — APPROVED** [review]
